lib/maths/common/CBayesianOptimisation.cc

```diff
@@ -114,9 +126,32 @@ CBayesianOptimisation::CBayesianOptimisation(core::CStateRestoreTraverser& trave
 }
 
 void CBayesianOptimisation::add(TVector x, double fx, double vx) {
-    m_FunctionMeanValues.emplace_back(x.cwiseQuotient(m_MaxBoundary - m_MinBoundary),
-                                      m_RangeScale * (fx - m_RangeShift));
-    m_ErrorVariances.push_back(CTools::pow2(m_RangeScale) * vx);
+    if (CMathsFuncs::isFinite(fx) == false || CMathsFuncs::isFinite(vx) == false) {
+        LOG_ERROR(<< "Discarding point (" << x.transpose() << "," << fx << "," << vx << ")");
+        return;
+    }
+
+    x = this->to01(std::move(x));
+    fx = toScaled(m_RangeShift, m_RangeScale, fx);
+    vx = CTools::pow2(m_RangeScale) * vx;
+
+    std::size_t duplicate(std::find_if(m_FunctionMeanValues.begin(),
```

Comment:
This is interesting. Why do you expect there to be duplicates?

Reply:
I don't in current usage but I think this code should handle them, i.e. we shouldn't be fragile, and there is a natural way to handle this.